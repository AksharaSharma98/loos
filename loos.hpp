/*
  This file is part of LOOS.

  LOOS (Lightweight Object-Oriented Structure library)
  Copyright (c) 2008, Tod D. Romo, Alan Grossfield
  Department of Biochemistry and Biophysics
  School of Medicine & Dentistry, University of Rochester

  This package (LOOS) is free software: you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation under version 3 of the License.

  This package is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.

  You should have received a copy of the GNU General Public License
  along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/






#if !defined(LOOS_HPP)
#define LOOS_HPP

// These are common system includes that nearly anybody who uses LOOS
// will probably be including anyway...

#include <iostream>
#include <iomanip>
#include <ios>
#include <sstream>
#include <fstream>

#include <ctime>
#include <cmath>

#include <string>
#include <vector>
#include <algorithm>

#include <stdexcept>

#include <cassert>

// These are the LOOS-specific includes...

#include <loos/loos_defs.hpp>
#include <loos/exceptions.hpp>
#include <loos/utils.hpp>
#include <loos/utils_random.hpp>
#include <loos/utils_structural.hpp>

#include <loos/Kernel.hpp>
#include <loos/Parser.hpp>
#include <loos/Selectors.hpp>


#include <loos/Matrix44.hpp>
#include <loos/XForm.hpp>
#include <loos/Matrix.hpp>

#include <loos/AtomicNumberDeducer.hpp>
#include <loos/Atom.hpp>
#include <loos/AtomicGroup.hpp>
#include <loos/pdb.hpp>
#include <loos/psf.hpp>
#include <loos/amber.hpp>
#include <loos/tinkerxyz.hpp>

#include <loos/Trajectory.hpp>
#include <loos/dcd.hpp>
#include <loos/dcd_utils.hpp>

#include <loos/trajwriter.hpp>
#include <loos/dcdwriter.hpp>
#include <loos/xtcwriter.hpp>

#include <loos/amber_traj.hpp>

#if defined(HAS_NETCDF)
#include <loos/amber_netcdf.hpp>
#endif

#include <loos/amber_rst.hpp>
#include <loos/ccpdb.hpp>
#include <loos/pdbtraj.hpp>
#include <loos/tinker_arc.hpp>
#include <loos/xtc.hpp>
#include <loos/gro.hpp>
#include <loos/trr.loos/hpp>



#include <loos/Geometry.hpp>
#include <loos/ensembles.hpp>
#include <loos/TimeSeries.hpp>

#include <loos/Fmt.hpp>

#include <loos/sfactories.hpp>

#include <loos/timer.hpp>
#include <loos/ProgressCounters.hpp>
#include <loos/ProgressTriggers.hpp>

#include <loos/sorting.hpp>

#include <loos/OptionsFramework.hpp>

#endif


