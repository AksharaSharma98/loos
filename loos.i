%module loos


%feature("autodoc", "1");

%include <std_string.i>
%include <std_vector.i>
%include <boost_shared_ptr.i>

typedef unsigned int    uint;
typedef unsigned long     ulong;

namespace loos {
  typedef double greal;
  typedef long gint;

  typedef float dcd_real;
  typedef double dcd_double;

  typedef Coord<double> GCoord;
}

%include "Coord.i"
%include "Atom.i"
%include "Matrix44.i"
%include "pdb_remarks.i"
%include "XForm.i"
%include "AtomicGroup.i"
%include "Trajectory.i"
%include "utils.i"
%include "cryst.i"
%include "pdb.i"
%include "sfactories.i"
%include "dcdwriter.i"
%include "ensembles.i"
%include "Geometry.i"
%include "TimeSeries.i"
%include "HBondDetector.i"
%include "exceptions.i"
