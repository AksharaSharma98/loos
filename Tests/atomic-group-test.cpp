/*
  This file is part of LOOS.

  LOOS (Lightweight Object-Oriented Structure library)
  Copyright (c) 2008, Tod D. Romo
  Department of Biochemistry and Biophysics
  School of Medicine & Dentistry, University of Rochester

  This package (LOOS) is free software: you can redistribute it and/or modify
  it under the terms of the GNU General Public License as published by
  the Free Software Foundation under version 3 of the License.

  This package is distributed in the hope that it will be useful,
  but WITHOUT ANY WARRANTY; without even the implied warranty of
  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
  GNU General Public License for more details.

  You should have received a copy of the GNU General Public License
  along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/


#include <iostream>
#include <string>
#include <vector>
#include <tr1/memory>


using namespace std;
using namespace tr1;

#include <loos.hpp>
#include <Atom.hpp>
#include <AtomicGroup.hpp>


struct SelectorCA : public AtomSelector {
  bool operator()(const pAtom& atom) const {
    return(atom->name() == "CA");
  }
};
  


int main() {
  pAtom a(new Atom(1, "CA", GCoord(1,2,3)));
  a->resid(1);
  pAtom b(new Atom(2, "CB", GCoord(4,5,6)));
  b->resid(1);
  pAtom c(new Atom(3, "CG", GCoord(-1, -2, -3)));
  c->resid(1);

  pAtom d(new Atom(4, "CA", GCoord(7,3,4)));
  d->resid(2);
  pAtom e(new Atom(5, "CG", GCoord(1,7,2)));
  e->resid(2);
  pAtom f(new Atom(4, "C", GCoord(1,7,2)));
  f->resid(2);

  a->addBond(b);
  a->addBond(c);
  cout << "Atom a:\n" << *a << endl;

  AtomicGroup g1;


  g1.append(a);
  g1.append(b);
  g1.append(c);
  g1.append(d);
  g1.append(e);

  cout << g1 << endl;
  cout << "-------------------\n";
  cout << "Operator= test:\n";
  g1 = g1;
  cout << "operator== test:\n";
  AtomicGroup ggg = g1;
  cout << (ggg == g1) << endl;
  cout << "Operator[] test:\n";
  pAtom tmpatm = g1[3];
  g1[3] = f;
  cout << g1 << endl;
  cout << "=====\n";
  g1[3] = tmpatm;
  cout << g1 << endl;

  cout << "-------------------\n";
  cout << "Operator+ tests:\n";
  AtomicGroup gg;
  gg.append(a);
  gg += b+c;
  cout << gg << endl;
  cout << "======\n";
  gg += g1;
  cout << gg << endl;
  cout << "=====\n";
  gg = a+b+c+d+e+f;
  cout << gg << endl;

  cout << "-------------------\n";
  cout << "CA selection:\n";
  SelectorCA sel;

  AtomicGroup s = g1.select(sel);
  cout << s << endl;



  cout << "-------------------\n";
  cout << "Iterator test:\n";
  AtomicGroup::Iterator ii(g1);
  pAtom pa;
  while (pa = ii())
    cout << *pa << endl;

  cout << "-------------------\n";
  cout << "Bounds test...\n";
  vector<GCoord> bb = g1.boundingBox();
  cout << bb[0] << " x " << bb[1] << endl;

  cout << "-------------------\n";
  cout << "Radius = " << g1.radius() << endl;
  cout << "Rgyr = " << g1.radiusOfGyration() << endl;

  cout << "-------------------\n";
  cout << g1.numberOfResidues() << " Residues in group.\n";
  AtomicGroup g2 = g1.getResidue(b);
  cout << "Size = " << g2.size() << endl;
  cout << g2 << endl;

  cout << "-------------------\n";
  AtomicGroup g4 = g1.copy();
  cout << "Clone & sharing test:\n";
  g4[0]->resid(999);
  cout << g4 << endl;
  cout << "-" << endl;
  cout << g1 << endl;

  cout << "---\n";
  AtomicGroup g5 = g4;
  g5[0]->resid(111);
  cout << g4 << endl;

  cout << "-------------------\n";
  cout << "subset(1,2):\n";
  AtomicGroup g3 = g1.subset(1,2);
  cout << g3 << endl;
  cout << "\nsubset(-2):\n";
  g3 = g1.subset(-2);
  cout << g3 << endl;
  cout << "\nsubset(-2,4):\n";
  g3 = g1.subset(-2,2);
  cout << g3 << endl;

  cout << "-------------------\n";
  cout << "Excise(1,2):\n";
  g3 = g1.excise(1,2);
  cout << g3 << endl << g1 << endl;

  cout << "-------------------\n";
  cout << "Box test:\n";
  AtomicGroup g7 = g1;
  g1.periodicBox(GCoord(13, 26, 39));
  cout << g1 << endl;

  cout << "-------------------\n";
  cout << "Box inheritance test:\n";
  cout << g3 << endl;
  cout << "Test updating of box to (25,26,29):\n";
  g1.periodicBox(25,26,29);
  cout << g1.periodicBox() << endl;
  cout << g3.periodicBox() << endl;
  cout << "Test updating derived box (pre-box) to (7,8,9):\n";
  g7.periodicBox(7,8,9);
  cout << g1.periodicBox() << endl;
  cout << "Testing copy (should be (7,8,9):\n";
  AtomicGroup g6 = g1.copy();
  g1.periodicBox(1,2,3);
  cout << g6.periodicBox() << endl;

}
