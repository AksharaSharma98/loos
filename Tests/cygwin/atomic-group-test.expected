Atom a:
<ATOM ID='1' NAME='CA' RESID='1' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
<GROUP>
   <ATOM ID='1' NAME='CA' RESID='1' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
   <ATOM ID='2' NAME='CB' RESID='1' RESNAME='   ' COORDS='(4,5,6)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='4' NAME='CA' RESID='2' RESNAME='   ' COORDS='(7,3,4)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='5' NAME='CG' RESID='2' RESNAME='   ' COORDS='(1,7,2)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>
-------------------
Operator= test:
operator== test:
1
Operator[] test:
<GROUP>
   <ATOM ID='1' NAME='CA' RESID='1' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
   <ATOM ID='2' NAME='CB' RESID='1' RESNAME='   ' COORDS='(4,5,6)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='4' NAME='C' RESID='2' RESNAME='   ' COORDS='(1,7,2)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='5' NAME='CG' RESID='2' RESNAME='   ' COORDS='(1,7,2)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>
=====
<GROUP>
   <ATOM ID='1' NAME='CA' RESID='1' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
   <ATOM ID='2' NAME='CB' RESID='1' RESNAME='   ' COORDS='(4,5,6)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='4' NAME='CA' RESID='2' RESNAME='   ' COORDS='(7,3,4)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='5' NAME='CG' RESID='2' RESNAME='   ' COORDS='(1,7,2)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>
-------------------
Operator+ tests:
<GROUP>
   <ATOM ID='1' NAME='CA' RESID='1' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
   <ATOM ID='2' NAME='CB' RESID='1' RESNAME='   ' COORDS='(4,5,6)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>
======
<GROUP>
   <ATOM ID='1' NAME='CA' RESID='1' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
   <ATOM ID='2' NAME='CB' RESID='1' RESNAME='   ' COORDS='(4,5,6)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='1' NAME='CA' RESID='1' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
   <ATOM ID='2' NAME='CB' RESID='1' RESNAME='   ' COORDS='(4,5,6)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='4' NAME='CA' RESID='2' RESNAME='   ' COORDS='(7,3,4)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='5' NAME='CG' RESID='2' RESNAME='   ' COORDS='(1,7,2)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>
=====
<GROUP>
   <ATOM ID='1' NAME='CA' RESID='1' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
   <ATOM ID='2' NAME='CB' RESID='1' RESNAME='   ' COORDS='(4,5,6)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='4' NAME='CA' RESID='2' RESNAME='   ' COORDS='(7,3,4)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='5' NAME='CG' RESID='2' RESNAME='   ' COORDS='(1,7,2)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='4' NAME='C' RESID='2' RESNAME='   ' COORDS='(1,7,2)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>
-------------------
CA selection:
<GROUP>
   <ATOM ID='1' NAME='CA' RESID='1' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
   <ATOM ID='4' NAME='CA' RESID='2' RESNAME='   ' COORDS='(7,3,4)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>
-------------------
Iterator test:
<ATOM ID='1' NAME='CA' RESID='1' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
<ATOM ID='2' NAME='CB' RESID='1' RESNAME='   ' COORDS='(4,5,6)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
<ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
<ATOM ID='4' NAME='CA' RESID='2' RESNAME='   ' COORDS='(7,3,4)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
<ATOM ID='5' NAME='CG' RESID='2' RESNAME='   ' COORDS='(1,7,2)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
-------------------
Bounds test...
(-1,-2,-3) x (7,7,6)
-------------------
Radius = 8.10679
Rgyr = 5.10686
-------------------
2 Residues in group.
Size = 3
<GROUP>
   <ATOM ID='2' NAME='CB' RESID='1' RESNAME='   ' COORDS='(4,5,6)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='1' NAME='CA' RESID='1' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
   <ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>
-------------------
Clone & sharing test:
<GROUP>
   <ATOM ID='1' NAME='CA' RESID='999' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
   <ATOM ID='2' NAME='CB' RESID='1' RESNAME='   ' COORDS='(4,5,6)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='4' NAME='CA' RESID='2' RESNAME='   ' COORDS='(7,3,4)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='5' NAME='CG' RESID='2' RESNAME='   ' COORDS='(1,7,2)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>
-
<GROUP>
   <ATOM ID='1' NAME='CA' RESID='1' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
   <ATOM ID='2' NAME='CB' RESID='1' RESNAME='   ' COORDS='(4,5,6)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='4' NAME='CA' RESID='2' RESNAME='   ' COORDS='(7,3,4)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='5' NAME='CG' RESID='2' RESNAME='   ' COORDS='(1,7,2)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>
---
<GROUP>
   <ATOM ID='1' NAME='CA' RESID='111' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
   <ATOM ID='2' NAME='CB' RESID='1' RESNAME='   ' COORDS='(4,5,6)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='4' NAME='CA' RESID='2' RESNAME='   ' COORDS='(7,3,4)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='5' NAME='CG' RESID='2' RESNAME='   ' COORDS='(1,7,2)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>
-------------------
subset(1,2):
<GROUP>
   <ATOM ID='2' NAME='CB' RESID='1' RESNAME='   ' COORDS='(4,5,6)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>

subset(-2):
<GROUP>
   <ATOM ID='1' NAME='CA' RESID='1' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
   <ATOM ID='2' NAME='CB' RESID='1' RESNAME='   ' COORDS='(4,5,6)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='4' NAME='CA' RESID='2' RESNAME='   ' COORDS='(7,3,4)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>

subset(-2,4):
<GROUP>
   <ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='4' NAME='CA' RESID='2' RESNAME='   ' COORDS='(7,3,4)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>
-------------------
Excise(1,2):
<GROUP>
   <ATOM ID='2' NAME='CB' RESID='1' RESNAME='   ' COORDS='(4,5,6)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>
<GROUP>
   <ATOM ID='1' NAME='CA' RESID='1' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
   <ATOM ID='4' NAME='CA' RESID='2' RESNAME='   ' COORDS='(7,3,4)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='5' NAME='CG' RESID='2' RESNAME='   ' COORDS='(1,7,2)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>
-------------------
Box test:
<GROUP PERIODIC='(13,26,39)'>
   <ATOM ID='1' NAME='CA' RESID='1' RESNAME='   ' COORDS='(1,2,3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'>
  <BOND>2</BOND>
  <BOND>3</BOND>
</ATOM>
   <ATOM ID='4' NAME='CA' RESID='2' RESNAME='   ' COORDS='(7,3,4)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='5' NAME='CG' RESID='2' RESNAME='   ' COORDS='(1,7,2)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>
-------------------
Box inheritance test:
<GROUP PERIODIC='(13,26,39)'>
   <ATOM ID='2' NAME='CB' RESID='1' RESNAME='   ' COORDS='(4,5,6)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
   <ATOM ID='3' NAME='CG' RESID='1' RESNAME='   ' COORDS='(-1,-2,-3)' ALTLOC=' ' CHAINID=' ' ICODE='' SEGID='    ' B='0' Q='0' CHARGE='0' MASS='1' ATOMICNUMBER='-1'/>
</GROUP>
Test updating of box to (25,26,29):
(25,26,29)
(25,26,29)
Test updating derived box (pre-box) to (7,8,9):
(7,8,9)
Testing copy (should be (7,8,9):
(7,8,9)
